### ISSUE TITLE
Our own docs and other projects that use tailwind are unusable with 32px root font size.

### ISSUE THREAD

**Issue by jguddas** [tailwindlabs/tailwindcss.com#1656]
## Why is this a problem?

It makes many features unusable and violates WCAG.

> ### WCAG 1.4.4 Resize Text (Level AA):
> Except for [captions](https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html#dfn-captions) and [images of text](https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html#dfn-image-of-text), [text](https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html#dfn-text) can be resized without [assistive technology](https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html#dfn-assistive-technology) up to 200 percent without loss of content or functionality.

## How to reproduce

1. Go to your browser settings and change the font size from `16px` to `32px`
https://support.google.com/chrome/answer/96810#fontsize:~:text=option%20you%20want.-,Change%20font%20size,-%3A%20Next%20to

![image](https://github.com/tailwindlabs/tailwindcss/assets/25524993/22be230c-7c92-47c8-8069-ab93d50fd06f)

2. Open https://tailwindcss.com/docs
3. Try to search (the nav bar is blocking you)

![image](https://github.com/tailwindlabs/tailwindcss/assets/25524993/fe3a4d83-4cd7-4760-83d8-fe243b1e86ab)

## What is causing this issue

1. This has simply not been considers while building the website.
2. The breakpoints are based on `px` but the sizes of the elements are not static.
3. The theme `spacing` scales with the font size as they are in `rem` rather than `px`.

https://github.com/tailwindlabs/tailwindcss/blob/26c4cb3b5c0c1833feaea6db79d4d0c54be0f84a/stubs/config.full.js#L816-L852

## Solutions

…

**Comment by adamwathan** [comment]
Hey! We've historically used pixels for media queries because of a [bug in Safari < 15](https://adamwathan.me/dont-use-em-for-media-queries/) that caused media queries defined with relative units to behave incorrectly when zooming, but switching your config over to using `em` units for media queries solves the issue you've raised here:

```js
// tailwind.config.js
module.exports = {
  // ...
  theme: {
    screens: {}
  }
}
```

Here's a branch of the Tailwind docs I just deployed with this change that works the way I think you'd expect:

https://tailwindcss-com-git-em-media-queries-tailwindlabs.vercel.app/docs/installation

This is a change to our default configuration that's already planned for Tailwind CSS v4.0 but can't be done out of the box sooner because it's technically a breaking change.

So as long as you don't need to support Safari 14 or prefer the trade-off of seeing that weird behavior instead of the weird behavior outlined in this issue, I would recommend switching to `em` units for media queries in your own projects by customizing your config 👍  Then in v4.0 we'll make it the default for everyone.

Hope that helps!

**Comment by jguddas** [comment]
> Then in v4.0 we'll make it the default for everyone.

That's wonderful news @adamwathan.

What about the `spacing`, now your [example](https://tailwindcss-com-git-em-media-queries-tailwindlabs.vercel.app/docs/installation) isn't glitched anymore but it still is a bit funky.

When I change the font size the whole layout gets zoomed just like if I pressed <kbd>CTRL +</kbd>.  
As a user, I don't want to zoom by changing the font size, I primarily want to simply change the font size.

> The theme spacing scales with the font size as they are in rem rather than px.
https://github.com/tailwindlabs/tailwindcss/blob/26c4cb3b5c0c1833feaea6db79d4d0c54be0f84a/stubs/config.full.js#L816-L852

What's your opinion on this? 

It would be awesome if we could somehow bring more awareness to the user and a11y to the end user.
